openpgp/read.go

```diff
@@ -400,6 +400,9 @@ func CheckDetachedSignatureAndHash(keyring KeyRing, signed, signature io.Reader,
 		issuerKeyId = *sig.IssuerKeyId
 		hashFunc = sig.Hash
 		sigType = sig.SigType
+		if hashFunc == crypto.MD5 {
+			return nil, errors.SignatureError("insecure hash algorithm: MD5")
+		}
```

Comment:
As is, I don't think this will ever trigger since you also removed MD5 from `HashById`, so `signature.Parse()` will already fail with unknown algorithm. Since this is a more informative error (and we probably still want to be able to read the message, and so don't want parsing it to fail), I would keep all the mappings between MD5 and its ID and name, but still remove the import of `crypto/md5`, so that we are still guaranteed to be unable to verify it, but trigger this error message instead.

Reply:
Good point :+1:

Reply:
:+1: Thanks! I would also keep the name mappings (`nameOfHash`, `nameToHash`, `hashNames`), so that when verifying a cleartext message with `Hash: MD5`, you get this error instead of "unknown hash". Also, in the future where we add more insecure hash functions, you could use `nameOfHash` in this very error message.